### ISSUE TITLE
Error while trying to read /tmp/hsperfdata_systest : [Errno 21] Is a directory: '/tmp/hsperfdata_systest'

### ISSUE THREAD

**Issue by haisum** [ansible/ansible#13987]
I am facing this issue randomly:

```
 [WARNING]: error while trying to read /tmp/hsperfdata_systest : [Errno 21] Is
a directory: '/tmp/hsperfdata_systest'

fatal: [127.0.0.1] => Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 586, in _executor
    exec_rc = self._executor_internal(host, new_stdin)
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 668, in _executor_internal
    inject = self.get_inject_vars(host)
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 611, in get_inject_vars
    combined_cache = self.get_combined_cache()
  File "/usr/lib/python2.6/site-packages/ansible/runner/__init__.py", line 606, in get_combined_cache
    combined_cache = self.setup_cache.copy()
  File "/usr/lib/python2.6/site-packages/ansible/cache/__init__.py", line 54, in copy
    return dict([(k, v) for (k, v) in self.iteritems()])
  File "/usr/lib64/python2.6/_abcoll.py", line 357, in iteritems
    yield (key, self[key])
  File "/usr/lib/python2.6/site-packages/ansible/cache/__init__.py", line 35, in __getitem__
    return self._plugin.get(key)
  File "/usr/lib/python2.6/site-packages/ansible/cache/jsonfile.py", line 69, in get
    f.close()
UnboundLocalError: local variable 'f' referenced before assignment


FATAL: all hosts have already failed -- aborting

```

This goes away if I remove `/tmp/hsperfdata_systest` and re-run playbook but re-appears again at some point.

I checked directory space, it's 20 GB free.

**Comment by haisum** [comment]
On further analysis I found this was because of fact caching in ansible.cfg:

``` ini
[defaults]

host_key_checking = False
retry_files_enabled = False
fact_caching = jsonfile
fact_caching_connection = /tmp
fact_caching_timeout = 86400
forks = 50
log_path=/var/log/ansible.log
```

I removed all fact\* lines and error stopped.

**Comment by jimi-c** [comment]
This issue has been corrected in devel/2.0, so we're going to go ahead and close this now.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
